### ISSUE TITLE
Indexing CRAM requires file to be sorted by chromosome / position ?

### ISSUE THREAD

**Issue by Z-Zen** [samtools/samtools#1288]
#### Are you using the latest version of [samtools](https://github.com/samtools/samtools/releases/latest) and [HTSlib](https://github.com/samtools/htslib/releases/latest)? 
Yes. More precisely: `samtools v1.10-86-gba462f6` and `htslib 1.10.2-109-gdcd4b73`

 #### Please describe your environment.
- OS: `Linux 4.9.0-9-amd64`
- machine architecture: `x86_64`
- compiler: `Debian 6.3.0-18+deb9u1`

#### Please specify the steps taken to generate the issue, the command you are running, and the relevant output.
First of all, sorry if this question was asked but I tried to look if issues were opened before and I couldn't find anything related to CRAM and indexing. Also, I checked the release changes if there was anything about CRAM changes but couldn't find anything neither.

In short, with version 1.9 of samtools, I didn't have this issue to index CRAM files without them being sorted by chromosome or position. Is it not possible to index the CRAM files without sorting them now ?

```
samtools view -T hg38.fasta -C -S -@ 6 -o sample.cram sample.bam
samtools index sample.cram
```

produces

```
[E::cram_index_build_multiref] CRAM file is not sorted by chromosome / position
samtools index: failed to create index for "sample.cram": No such file or directory
```

**Comment by valeriuo** [comment]
It wasn't possible in 1.9 either. The [1.9 manual](http://www.htslib.org/doc/1.9/samtools.html) states under the index section:

> Index a coordinate-sorted BAM or CRAM file for fast random access.

**Comment by jkbonfield** [comment]
Technically CRAM can support limited indexing on non-positional data.  Io_lib's `cram_index` can do this.  However it is of limited use.

The primary purpose for such data is in chunking datasets up by the Nth container / slice.  Eg a CRAM file of unmapped reads where you wish to process it in parallel in blocks of 1 million reads (equiv to 100 slices with default params).  It appears I broke this (`cram_filter`) in the latest io_lib release though!  Bah...

Anyway this isn't something we support in htslib.  It's maybe worth considering it at some point though.

**Comment by Z-Zen** [comment]
> It wasn't possible in 1.9 either. The [1.9 manual](http://www.htslib.org/doc/1.9/samtools.html) states under the index section:
> 
> > Index a coordinate-sorted BAM or CRAM file for fast random access.

I am surprised because I am pretty sure we did run samtools 1.9 and htslib 1.9. From the log of old commandline:

```
samtools 1.9
Using htslib 1.9
Copyright (C) 2018 Genome Research Ltd.
```

In any case, thanks for the feedback.

**Comment by jkbonfield** [comment]
IGV isn't going to work on anything other than coordinate sorted data, so if it worked in the past then you've just misled yourself at best.

The indexing of non-sorted data is very limited in use, restricted mainly to a few specific parallel processing scenarios.  So disabling it until we have a viable per-slice or per-container subsetting utility in samtools/htslib seems reasonable.
